### PR TITLE
GSoC '16 Testing Project, prelim task: skip slow tests (not to be merged)

### PR DESCRIPTION
**Pull request by karandesai-96**
* This PR is for discussions related to my preliminary task for **Tardis Testing** project ([TEP001](https://github.com/tardis-sn/tep/blob/master/TEP001_extensive_test_suite.rst)).

* The task was to mark the full Tardis test as **slow** and skip it automatically during normal run of tests, for example at Travis CI, and include slow tests only while running on a **dedicated server**.

* The full Tardis test is marked as slow and will only run when `--run-slow` parameter is provided while running the tests. In a nutshell:
    * `./setup.py test` will skip full Tardis test inherently, no special parameters to provide.
    * `./setup.py test --run-slow` will include Tardis slow test while running.

* A brief description is provided as a note in the _diffs_ below, please refer to it.

### CODE REVIEW

File: tardis/conftest.py [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 
 def pytest_addoption(parser):
+    parser.addoption("--run-slow", action="store_true",
+                     help="include running slow tests during run")
+
```
Comment (karandesai-96) [inline tardis/conftest.py]: An option for detecting the command line flag `--run-slow` has to be added in **conftest.py.** The task of detecting it is silently handled by pytest behind the curtains. :smile:

File: tardis/tests/test_tardis_full.py [hunk 2]
```diff
@@ -20,6 +20,8 @@ def data_path(fname):
 
 @pytest.mark.skipif(not pytest.config.getvalue("atomic-dataset"),
                     reason='--atomic_database was not specified')
+@pytest.mark.skipif(not pytest.config.getoption("--run-slow"),
+                    reason='this is a slow test, add --run-slow to run')
```
Comment (karandesai-96) [inline tardis/tests/test_tardis_full.py]: **Skip if**  the flag `--run-slow` is **not** obtained by parsing the entered py.test command from command line.

**Comment by karandesai-96** [comment]
#### The slow integration test for Stratified W7 Setup now works fine!

- I have made a lot of assertions - I currently have very little idea about the quantities involved in calculation so I might have missed/ included redundant assertions. Current set of assertions made as loosely adapted from **test_tardis_full.py**.

- The current directory structure:
```
tardis
    └── tardis
        └── tests
            ├── data
            ├── w7_13d
            │   ├── tardis_w7.yml
            │   ├── w7_densitites.dat
            │   ├── w7_abundancies.dat
            │   ├── expected_ndarrays.npz
            │   ├── expected_quantities.npz
            │   └── w7_collect.py
            └── test_w7.py
```

-  **tardis_w7.yml, w7_densitites.dat, w7_abundancies.dat** were collected from https://github.com/tardis-sn/tep/pull/13

- A believed-stable version of TARDIS was used to execute a single run with the Stratified W7 setup and benchmark data was collected for further assertions.

- **expected_ndarrays.npz** is a collection of parameters in the `radial1d_model`, whose type is `numpy.ndarray`. The file size is 35.4 MB and it contains these parameters from the `radial1d_model`:
    - `last_interaction_type`
    - `last_line_interaction_out_id`
    - `last_line_interaction_in_id`
    - `j_estimators`
    - `j_blue_estimators`
    - `last_line_interaction_shell_id`
    - `nubar_estimators`
    - `ws`

- **expected_astropy_quantities.npz** is a collection of parameters in the `radial1d_model`, whose type is 'astropy.units.quantity.Quantity`, though only the value is stored and the unit of quantity is multiplied back before making assertions. The file size is 3.9 MB and it contains these parameters from the `radial1d_model`:
    - `t_rads`
    - `luminosity_inner`
    - `montecarlo_luminosity`
    - `montecarlo_virtual_luminosity`
    - `time_of_simulation`
    - `montecarlo_nu`
    - `last_line_interaction_angstrom`
    - `j_blues_norm_factor`

- I wanted to get the flow of writing a complete class for doing an integration test with one of TARDIS' "slow" configurations hence wrote down a separate class.

- I have tested my code - just one integration test and here is the screenshot after the final run:

image

**Comment by unoebauer** [comment]
@karandesai-96 - is this PR still needed? If not, we should close it.

**Comment by karandesai-96** [comment]
This PR had configuration files for AT setup as well, but I have the branch locally. Whatever building blocks were required, as already including in #557 and #558. Closing it finally.

**Pull request closed without merging**